Druid/DRAbstractInstruction.class.st

```diff
@@ -232,11 +232,11 @@ DRAbstractInstruction >> validate [
 	self validateDependenciesAreUsedByMyself.
 	self validateUsersUseMyself.
 
-	self isPhiFunction ifTrue: [ ^ self ].
 	self operands do: [ :o |
-		((o isInstruction not or: [
-			  self basicBlock isDominatedBy: o basicBlock ]) or: [ "this is to handle the case we have different CFG, block is interpreted with a new CFG"
-			 self controlFlowGraph ~= o controlFlowGraph ]) ifFalse: [
+		(o isInstruction not or: [
+			 (self isDominatedBy: o) or: [
+				 "this is to handle the case we have different CFG, block is interpreted with a new CFG"
```

Comment:
The comment is old not valid anymore no?

Reply:
Or is it about the code below?

When can we have instructions using operands from a different cfg? This looks broken semantics.
@PalumboN ideas?

Reply:
I have no idea, but for me each CFG is independent from others